### DISCUSSION TITLE
Alert notification are truncated / incomplete

### DISCUSSION THREAD

**Discussion by garceri** [grafana/grafana#37011] [Alerting]
**What happened**:
Got an alert configured for a set of about 100 servers, about 20 of them should alert as they meet the alert condition, only  6 of them show up on the Alert (using Telegram and Slack), however Alert State History show the full list of servers that triggered the alert.

**What you expected to happen**:
Alert should contain a complete list of failed servers
**How to reproduce it (as minimally and precisely as possible)**:
Setup alert to telegram/slack
Create Alert Panel with threshold low enough to cause more than 6 servers to fall below it and trigger the alert. 
**Anything else we need to know?**:

**Environment**:
- Grafana version: Grafana v6.2.0 Nightly 4/17/19
- Data source type & version: Prometheus 2.1.0
- OS Grafana is installed on: Ubuntu 18.04.1
- User OS & Browser: Chrome Windows x86_64 Latest Version
- Grafana plugins: None
- Others:
The prometheus query i'm using is something like this:
`avg(mem_free{instance =~ "rack1.*"}) by (instance)`

**Comment by torkelo** [comment]
Limit is here if anyone want to change it:
https://github.com/grafana/grafana/blob/master/pkg/services/alerting/notifiers/telegram.go#L186

**Comment by garceri** [comment]
Mnhh, the limit on that file is 4, yet im getting 6 entries, is that normal or what does that 4 mean ?

**Comment by garceri** [comment]
@torkelo you sure that's the limit? As i stated before i'm getting 6 servers on the alert yet that value is set to 4 by default...
And by the way i get the same 6 servers on my Slack alert as well.

**Comment by garceri** [comment]
Any news on this one ?

**Comment by knmorgan** [comment]
@torkelo Any update on this? This is also impacting us.

The [slack notifier](https://github.com/grafana/grafana/blob/master/pkg/services/alerting/notifiers/slack.go#L201) seemingly sets a limit of four fields, but we are seeing six in our alerts. We would prefer to see everything that is currently alerting, however.

We're using Grafana Cloud so changing the code isn't an option for us anyway.

**Comment by lucasRolff** [comment]
We're having this issue as well. In some cases we'll have alerts triggering for 20+ machines at once, we'd like to show more than just 6 hostnames in Slack. I'm aware there may be an upper limit of the APIs at some point but 6 seems a bit low.

**Comment by mator** [comment]
Anyone with more recent grafana, like 7.x or 8.x ? does it still truncates telegram messages? Thanks.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
